machine-learning/01-medulloblastoma_data_prep.Rmd

```diff
@@ -0,0 +1,272 @@
+---
+title: "Northcott Medulloblastoma: Data preparation"
+output:   
+  html_notebook: 
+    toc: true
+    toc_float: true
+---
+
+**J. Taroni and C. Greene 2018**
+
+In this notebook, we'll cover how to do conversion between different gene
+identifiers.
+Different tools use different ids.
+[`PLIER`](https://github.com/wgmao/PLIER) 
+([Mao, et al. _bioRxiv._ 2017.](https://doi.org/10.1101/116061)), 
+which we'll use shortly, 
+requires that we use gene symbols and the refine.bio processed data uses
+Ensembl gene ids.
+
+For this part of the workshop, we'll be using [`GSE37382`](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE37382),
+primary medulloblastoma sample data from 
+[Northcott, et al. _Nature._ 2012.](https://www.ncbi.nlm.nih.gov/pubmed/22832581).
+
+## Libraries and functions
+
+We'll need something that maps between different types of identifiers to
+do the conversion.
+Here, we'll use the `org.Hs.eg.db` package from [Bioconductor]().
+
+For more information about using Bioconductor annotation packages, see 
+[this vignette](http://bioconductor.org/packages/release/bioc/vignettes/AnnotationDbi/inst/doc/IntroToAnnotationPackages.pdf) about AnnotationDbi.
+
+```{r}
+library(org.Hs.eg.db)
+```
+
+```{r}
+# magrittr pipe
+`%>%` <- dplyr::`%>%` 
+```
+
+
+## Files and directories
+
+```{r}
+# the directory where the refine.bio processed data lives, it's also where we 
+# will write the output after the conversion
+data_dir <- file.path("data", "GSE37382")
+# refine.bio processed expression file
+exprs_file <- file.path(data_dir, "GSE37382_SCAN.pcl")
+# the file used for mapping between gene ids that is output
+mapping_file <- file.path(data_dir, 
+                          "genesymbol_ensembl_mappings_used_for_GSE37382.tsv")
+# the prepped expression data output fule
+output_file <- file.path(data_dir, "GSE37382_SCAN_symbol_mean_agg.pcl")
+```
+
+## Read in data
+
+We'll read in the data with a [`data.table`](https://cran.r-project.org/web/packages/data.table/index.html) function because it is much faster than base R.
+
+```{r}
+exprs_df <- data.table::fread(exprs_file, data.table = FALSE)
+```
+
+Let's take a look at the expression `data.frame`.
+
+```{r}
+exprs_df[1:5, 1:5]
+```
+
+We can see that the first column, called `Gene`, contains the Ensembl gene ids.
+Let's take a look at the dimensions of `exprs_df`.
+
+```{r}
+dim(exprs_df)
+```
+
+This means there are 21661 rows, which correspond to genes, and 285 samples
+because the first of the 286 columns contain gene identifers.
+
+## Annotation
+
+### Conversion
+
+`PLIER` uses gene symbols, but `org.Hs.eg.db` can be used for a number of 
+different conversions. 
+Let's take a look at what it contains.
+
+```{r}
+keytypes(org.Hs.eg.db)
```

Comment:
Do we want to note that the `Hs` means human? It may be helpful to note that somewhere - perhaps here.

Reply:
I ended up noting it at the top of this notebook.